### PR TITLE
Implement helper to open reports with the viz app

### PR DESCRIPTION
**Pull request by folmos-at-orange**

### CODE REVIEW

File: khiops/utils/helpers.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    # Check that the report path has a valid file extension
+    _, ext = os.path.splitext(report_path)
+    if ext not in [".khj", ".khjc"]:
+        raise ValueError(
+            "'report_path' must has extension '.khj' or '.khcj'. "
+            f"Path: {report_path}"
+        )
+
+    # Open it with the associated application
+    try:
```
Comment (popescu-v) [inline khiops/utils/helpers.py:26]: I would use the `KhiopsRuntimeError` exception, and distinguish between:

- inexistent or inadequate (like: directory instead of file, or lack of reading rights) `report_path`
- inaccessible visualization tool / inability to open the report path  (as of now, but IMHO the error should be encapsulated in a `KhiopsRuntimeError`

Comment (folmos-at-orange) [reply khiops/utils/helpers.py:26]: This function is intended to be used in interactive mode (think of a jupyter notebook). It should not fail with an exception, simply with a message so as to not cut the program flow. The error is passed to the user simply as a message.

**Review by popescu-v — COMMENTED** [review]
Some minor (mostly of ergonomy) comments.

But IMHO before merging, we need to establish consensus on the desired behavior, among which:
1. Current behavior :
   - advantage : simple to implement and maintain; clear;
   - disadvantage : ergonomically less optimal, as users need to get the report file themselves (potentially from temporary directories), then to pass it to the helper.
2. Adding the helper as a method of the estimators:
    - advantage : improved ergonomy, the mere existence of the report file is transparent to the users
    - disadvantage: needing to decide what happens to the report file once the estimator object instance is destroyed: either keep it or delete it:
       -  if we delete it, then this is seems OK for Khiops, but not for Khiops Coclustering, where the report can be changed through the visualization tool; moreover, on Windows, we cannot delete the report file unless the visualization tool is also closed first (or a different file loaded into it).
       - if we keep it, then the estimators would, in general, leak their report files after their disappearance as objects.
       - if we first close the visualization tool, then delete the report files when the estimator object gets destroyed or reinitialized, then the code becomes more complex, because we need to decide whether: this is done when exiting the Python process; or, this is done when the estimator object disappears from the current scope (gets deleted via `del`, or for other reasons).

**Comment by folmos-at-orange** [comment]
> Some minor (mostly of ergonomy) comments.
> 
> But IMHO before merging, we need to establish consensus on the desired behavior, among which:
> 
> 1. Current behavior :
>    
>    * advantage : simple to implement and maintain; clear;
>    * disadvantage : ergonomically less optimal, as users need to get the report file themselves (potentially from temporary directories), then to pass it to the helper.

There is no requirement for sklearn users to use a temporary directory. Usually they'd simply export to their current directory
```python
khc.export_report_file("report.khj")
open_report("report.khj")
```

> 2. Adding the helper as a method of the estimators:
>    
>    * advantage : improved ergonomy, the mere existence of the report file is transparent to the users
>    * disadvantage: needing to decide what happens to the report file once the estimator object instance is destroyed: either keep it or delete it:
>      
>      * if we delete it, then this is seems OK for Khiops, but not for Khiops Coclustering, where the report can be changed through the visualization tool; moreover, on Windows, we cannot delete the report file unless the visualization tool is also closed first (or a different file loaded into it).
>      * if we keep it, then the estimators would, in general, leak their report files after their disappearance as objects.
>      * if we first close the visualization tool, then delete the report files when the estimator object gets destroyed or reinitialized, then the code becomes more complex, because we need to decide whether: this is done when exiting the Python process; or, this is done when the estimator object disappears from the current scope (gets deleted via `del`, or for other reasons).

**Comment by popescu-v** [comment]
> > Some minor (mostly of ergonomy) comments.
> > But IMHO before merging, we need to establish consensus on the desired behavior, among which:
> > 
> > 1. Current behavior :
> >    
> >    * advantage : simple to implement and maintain; clear;
> >    * disadvantage : ergonomically less optimal, as users need to get the report file themselves (potentially from temporary directories), then to pass it to the helper.
> 
> There is no requirement for sklearn users to use a temporary directory. Usually they'd simply export to their current directory
> 
> ```python
> khc.export_report_file("report.khj")
> open_report("report.khj")
> ```
> 

Right. It seems rather straightforward after all.

> > 2. Adding the helper as a method of the estimators:
> >    
> >    * advantage : improved ergonomy, the mere existence of the report file is transparent to the users
> >    * disadvantage: needing to decide what happens to the report file once the estimator object instance is destroyed: either keep it or delete it:
> >      
> >      * if we delete it, then this is seems OK for Khiops, but not for Khiops Coclustering, where the report can be changed through the visualization tool; moreover, on Windows, we cannot delete the report file unless the visualization tool is also closed first (or a different file loaded into it).
> >      * if we keep it, then the estimators would, in general, leak their report files after their disappearance as objects.
> >      * if we first close the visualization tool, then delete the report files when the estimator object gets destroyed or reinitialized, then the code becomes more complex, because we need to decide whether: this is done when exiting the Python process; or, this is done when the estimator object disappears from the current scope (gets deleted via `del`, or for other reasons).

File: doc/samples/samples.rst [hunk 2]
```diff
@@ -1223,14 +1226,17 @@ Samples
...
-    coclustering_file_path = kh.train_coclustering(
+    coclustering_report_path = kh.train_coclustering(
         dictionary_file_path,
         "SpliceJunctionDNA",
         data_table_path,
         ["SampleId", "Char"],
         results_dir,
     )
-    print("Coclustering file available at " + coclustering_file_path)
+    print(f"Coclustering file available at {coclustering_report_path}")
```
Comment (popescu-v) [inline doc/samples/samples.rst:1236]: I would rephrase to "Coclustering report file available at ..."

File: doc/samples/samples.rst [hunk 3]
```diff
@@ -1223,14 +1226,17 @@ Samples
...
+
+    # If you have Khiops Co-Visualization app you may open the report with the following
```
Comment (popescu-v) [inline doc/samples/samples.rst:1238]: "If you have **the** Khiops Co-Visualization app..."

File: doc/samples/samples_sklearn.rst [hunk 4]
```diff
@@ -94,6 +94,10 @@ Samples
     test_auc = metrics.roc_auc_score(y_test, y_test_probas[:, 1])
     print(f"Test accuracy = {test_accuracy}")
     print(f"Test auc      = {test_auc}")
+
+    # If you have Khiops Visualization app you may open the report with the following
```
Comment (popescu-v) [inline doc/samples/samples_sklearn.rst:98]: "If you have **the** Khiops Visualization app..."

File: doc/samples/samples_sklearn.rst [hunk 5]
```diff
@@ -520,6 +524,10 @@ Samples
     test_mae = metrics.mean_absolute_error(y_test, y_test_pred)
     print(f"Test R2  = {test_r2}")
     print(f"Test MAE = {test_mae}")
+
+    # If you have Khiops Visualization app you may open the report with the following
```
Comment (popescu-v) [inline doc/samples/samples_sklearn.rst:528]: Add definite article "the" before "Khiops" (see also above): this should be done wherever applicable.

File: doc/samples/samples_sklearn.rst [hunk 6]
```diff
@@ -549,6 +557,10 @@ Samples
     print("Encoded data:")
     print(X_transformed[:10])
     print("---")
+
+    # If you have Khiops Visualization app you may open the report with the following
```
Comment (popescu-v) [inline doc/samples/samples_sklearn.rst:561]: Add definite article (see other comments above).

File: doc/samples/samples_sklearn.rst [hunk 7]
```diff
@@ -802,6 +814,10 @@ Samples
     print("Predicted clusters (first 10)")
     print(X_clusters[:10])
     print("---")
+
+    # If you have Khiops Co-Visualization app you may open the report with the following
```
Comment (popescu-v) [inline doc/samples/samples_sklearn.rst:818]: Definite article is missing (see above).

File: khiops/core/helpers.py [hunk 8]
```diff
@@ -444,3 +446,53 @@ def deploy_predictor_for_metrics(
 
 
 # pylint: enable=protected-access
+
+
+def visualize_report(report_path):
+    """Opens a Khiops or Khiops Coclustering report with the desktop visualization app
+
+    Before using this function, make sure you have installed the Khiops visualization
+    and Khiops Co-Visualization apps.  More info at
```
Comment (popescu-v) [inline khiops/core/helpers.py:455]: I would replace the "and" by "and / or", because some users could only be interested in one of the two kinds of reports: khiops or coclustering.

File: khiops/samples/samples.py [hunk 9]
```diff
@@ -229,6 +229,9 @@ def train_predictor_file_paths():
     print("Reports file available at " + report_file_path)
     print("Modeling dictionary file available at " + modeling_dictionary_file_path)
 
+    # If you have Khiops Visualization app you may open the report with the following
```
Comment (popescu-v) [inline khiops/samples/samples.py:232]: Definite article missing (see comments above).

File: khiops/samples/samples.py [hunk 10]
```diff
@@ -1376,14 +1379,17 @@ def train_coclustering():
...
-    coclustering_file_path = kh.train_coclustering(
+    coclustering_report_path = kh.train_coclustering(
         dictionary_file_path,
         "SpliceJunctionDNA",
         data_table_path,
         ["SampleId", "Char"],
         results_dir,
     )
-    print("Coclustering file available at " + coclustering_file_path)
+    print(f"Coclustering file available at {coclustering_report_path}")
```
Comment (popescu-v) [inline khiops/samples/samples.py:1389]: "Coclustering **report** file available at ..."

File: khiops/samples/samples.py [hunk 11]
```diff
@@ -1376,14 +1379,17 @@ def train_coclustering():
...
+
+    # If you have Khiops Co-Visualization app you may open the report with the following
```
Comment (popescu-v) [inline khiops/samples/samples.py:1391]: Add definite article.

File: khiops/samples/samples_sklearn.ipynb [hunk 12]
```diff
@@ -79,7 +79,11 @@
     "test_accuracy = metrics.accuracy_score(y_test, y_test_pred)\n",
     "test_auc = metrics.roc_auc_score(y_test, y_test_probas[:, 1])\n",
     "print(f\"Test accuracy = {test_accuracy}\")\n",
-    "print(f\"Test auc      = {test_auc}\")"
+    "print(f\"Test auc      = {test_auc}\")\n",
+    "\n",
+    "# If you have Khiops Visualization app you may open the report with the following\n",
```
Comment (popescu-v) [inline khiops/samples/samples_sklearn.ipynb:84]: Add definite article.

**Review by popescu-v — CHANGES_REQUESTED** [review]
See comments.

File: khiops/samples/samples.py (hunk 11 above)
Comment (folmos-at-orange) [reply khiops/samples/samples.py:1391]: Ok & Done

File: khiops/core/helpers.py (hunk 8 above)
Comment (folmos-at-orange) [reply khiops/core/helpers.py:455]: Ok

Comment (folmos-at-orange) [reply khiops/core/helpers.py:455]: Done.

File: khiops/samples/samples.py [hunk 13]
```diff
@@ -229,6 +229,9 @@ def train_predictor_file_paths():
     print("Reports file available at " + report_file_path)
     print("Modeling dictionary file available at " + modeling_dictionary_file_path)
 
+    # If you have the Khiops Visualization app you may open the report as follows
```
Comment (popescu-v) [inline khiops/samples/samples.py:232]: I would update, everywhere needed, to:

```python
# If you have the Khiops Visualization application installed on your local machine, you may open the report as follows
```

It is longer, but clearer IMHO, by adding the two key points: "installed" (it's not a ready-to-use executable, but a package that needs to be installed) and "on your local machine" (it's not a remote service to be configured, or a client to a remote service, and neither is it installable in a virtualenv or conda).

File: khiops/core/helpers.py [hunk 14]
```diff
@@ -444,3 +446,53 @@ def deploy_predictor_for_metrics(
...
+    # - is not executable
+    _, ext = os.path.splitext(report_path)
+    if ext not in [".khj", ".khcj"]:
+        raise ValueError(
+            "'report_path' must have extension '.khj' or '.khcj'. "
+            f"Path: {report_path}"
+        )
+    if not os.path.exists(report_path):
+        raise FileNotFoundError(report_path)
+    if platform.system() != "Windows" and os.access(report_path, os.X_OK):
```
Comment (popescu-v) [inline khiops/core/helpers.py:485]: I would add a comment wrt. the supplemental Windows guard.

**Review by popescu-v — CHANGES_REQUESTED** [review]
2 comments.

File: khiops/samples/samples_sklearn.py [hunk 15]
```diff
@@ -525,6 +529,10 @@ def khiops_regressor():
     print(f"Test R2  = {test_r2}")
     print(f"Test MAE = {test_mae}")
 
+    # If you have the Khiops Visualization app you may open the report as follows
```
Comment (popescu-v) [inline khiops/samples/samples_sklearn.py:532]: Add "installed".

File: khiops/samples/samples_sklearn.py [hunk 16]
```diff
@@ -564,6 +572,10 @@ def khiops_encoder():
     print(X_transformed[:10])
     print("---")
 
+    # If you have the Khiops Visualization app you may open the report as follows
```
Comment (popescu-v) [inline khiops/samples/samples_sklearn.py:575]: Add "installed".

File: khiops/samples/samples_sklearn.py [hunk 17]
```diff
@@ -833,6 +845,10 @@ def khiops_coclustering():
     print(X_clusters[:10])
     print("---")
 
+    # If you have the Khiops Co-Visualization app you may open the report as follows
```
Comment (popescu-v) [inline khiops/samples/samples_sklearn.py:848]: Add "installed".

**Review by popescu-v — APPROVED** [review]
LGTM
